userspace/libsinsp/async_cgroup.cpp

```diff
@@ -0,0 +1,97 @@
+#include "async_cgroup.h"
+
+#include <fstream>
+#include "sinsp.h"
+
+namespace {
+// to prevent 32-bit number of kilobytes from overflowing, ignore values larger than 4 TiB.
+// This reports extremely large values (e.g. almost-but-not-quite 9EiB as set by k8s) as unlimited.
+// Note: we use the same maximum value for cpu shares/quotas as well; the typical values are much lower
+// and so should never exceed CGROUP_VAL_MAX either
+constexpr const int64_t CGROUP_VAL_MAX = 1UL << 42u;
+
+bool read_cgroup_val(std::shared_ptr<std::string>& subsys, const std::string& cgroup, const std::string& filename, int64_t& out)
+{
+	if(cgroup == "/")
+	{
+		return false;
+	}
+
+	std::string path = *subsys.get() + "/" + cgroup + "/" + filename;
+	std::ifstream cg_val(path);
+
+	int64_t val;
+	cg_val >> val;
+
+	if(val > 0 && val < CGROUP_VAL_MAX)
+	{
+		out = val;
+		return true;
+	}
+	return false;
+}
+}
+
+namespace libsinsp {
+namespace async_cgroup {
+
+bool get_cgroup_resource_limits(const delayed_cgroup_key& key, delayed_cgroup_value& value)
+{
+	bool found_all;
+
+	std::shared_ptr<std::string> memcg_root = sinsp::lookup_cgroup_dir("memory");
+	g_logger.format(sinsp_logger::SEV_DEBUG, "(async-cg) mem cgroup for container [%s]: %s/%s",
+		key.m_container_id.c_str(), memcg_root->c_str(), key.m_mem_cgroup.c_str());
+	found_all = read_cgroup_val(memcg_root, key.m_mem_cgroup, "memory.limit_in_bytes", value.m_memory_limit);
```

Comment:
I'd be tempted to initialize `found_all` to `true` and use the same pattern here as you've used below.  That way, if for any reason things get reordered in the future we don't end up with an unintentional bug.